<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from arm.xacro                      | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<!-- Revolute-Revolute Manipulator -->
<robot name="my_robotic_arm" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <material name="black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="blue">
    <color rgba="0.0 0.0 0.8 1.0"/>
  </material>
  <material name="green">
    <color rgba="0.0 0.8 0.0 1.0"/>
  </material>
  <material name="darkgreen">
    <color rgba="0.0 0.3 0.0 1.0"/>
  </material>
  <material name="gray">
    <color rgba="0.7 0.7 0.7 1.0"/>
  </material>
  <material name="darkgray">
    <color rgba="0.3 0.3 0.3 1.0"/>
  </material>
  <material name="red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="white">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <material name="yellow">
    <color rgba="0.8 0.8 0.0 1.0"/>
  </material>
  <!-- Used for fixing robot to Gazebo 'link 1' -->
  <link name="world"/>
  <joint name="j1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="world"/>
    <child link="joint1"/>
    <axis xyz="0 0 1"/>
    <limit effort="100" lower="-2.35619449019" upper="2.35619449019" velocity="5.0"/>
    <dynamics damping="0.7"/>
  </joint>
  <link name="joint1">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.0"/>
      <geometry>
        <cylinder length="0.2" radius="0.05"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.0"/>
      <geometry>
        <cylinder length="0.2" radius="0.05"/>
      </geometry>
    </visual>
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 0.00000 0.0"/>
      <mass value="0.5"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <!--    <joint name="link1_fix" type="fixed">
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <parent link="joint1"/>
        <child link="link1"/>
    </joint>
    <link name="link1">
        <collision>
            <origin xyz="0 0 ${l1/2}" rpy="0 0 0"/>
            <geometry>
                <box size="${width} ${width} ${l1}" />
            </geometry>
        </collision>
        <visual>
            <origin xyz="0 0 ${l1/2}" rpy="0 0 0"/>
            <geometry>
                <box size="${width} ${width} ${l1}"/>
            </geometry>
            <material name="darkgreen"/>
        </visual>
        <inertial>
            <origin xyz="0.0 0.00000 0.0" rpy="0 0 0"/>
            <mass value="0.5"/>
            <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0" />
        </inertial>
    </link>
   -->
  <joint name="j2" type="revolute">
    <origin rpy="1.57079632679 0.0 1.57079632679" xyz="0 0 0.3"/>
    <parent link="joint1"/>
    <child link="joint2"/>
    <axis xyz="0 0 1 "/>
    <limit effort="100" lower="-2.35619449019" upper="2.35619449019" velocity="5.0"/>
    <dynamics damping="0.7"/>
  </joint>
  <link name="joint2">
    <collision>
      <origin rpy="0 -0.0 0" xyz=" 0 0 0.0"/>
      <geometry>
        <cylinder length="0.2" radius="0.05"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 -0.0 0" xyz="0 0 0.0 "/>
      <geometry>
        <cylinder length="0.2" radius="0.05"/>
      </geometry>
    </visual>
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 0.00000 0.0"/>
      <mass value="0.5"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <!--    <joint name="link2_fix" type="fixed">
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <parent link="joint2"/>
        <child link="link2"/>
    </joint>
    <link name="link2">
        <collision>
            <origin xyz="0 ${l2/2} 0 " rpy="0 0 0"/>
            <geometry>
                <box size="${width} ${l2} ${width}" />
            </geometry>
        </collision>
        <visual>
            <origin xyz="0 ${l2/2} 0 " rpy="0 0 0"/>
            <geometry>
                <box size="${width} ${l2} ${width}"/>
            </geometry>
            <material name="darkgreen"/>
        </visual>
        <inertial>
            <origin xyz="0.0 0.00000 0.0" rpy="0 0 0"/>
            <mass value="0.5"/>
            <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0" />
        </inertial>
    </link>
    -->
  <joint name="j3" type="revolute">
    <origin rpy="0.0 0.0 1.57079632679" xyz="0 0.3 0"/>
    <parent link="joint2"/>
    <child link="joint3"/>
    <axis xyz="0 0 1"/>
    <limit effort="100" lower="-2.35619449019" upper="2.35619449019" velocity="5.0"/>
    <dynamics damping="0.7"/>
  </joint>
  <link name="joint3">
    <collision>
      <origin rpy="0 -0.0 0" xyz="0 0.0 0"/>
      <geometry>
        <cylinder length="0.2" radius="0.05"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 -0.0 0" xyz="0 0.0  0"/>
      <geometry>
        <cylinder length="0.2" radius="0.05"/>
      </geometry>
    </visual>
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 0.00000 0.0"/>
      <mass value="0.5"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <!--    <joint name="link3_fix" type="fixed">
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <parent link="joint3"/>
        <child link="link3"/>
    </joint>
    <link name="link3">
        <collision>
            <origin xyz="${l3/2} 0 0 " rpy="0 0 0"/>
            <geometry>
                <box size="${l3} ${width} ${width}" />
            </geometry>
        </collision>
        <visual>
            <origin xyz="${l3/2} 0 0 " rpy="0 0 0"/>
            <geometry>
                <box size="${l3} ${width} ${width}"/>
            </geometry>
            <material name="darkgreen"/>
        </visual>
        <inertial>
            <origin xyz="0.0 0.00000 0.0" rpy="0 0 0"/>
            <mass value="0.5"/>
            <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0" />
        </inertial>
    </link>
    
    -->
  <joint name="j4" type="revolute">
    <origin rpy="1.57079632679 0.0 1.57079632679" xyz="0.3 0 0"/>
    <parent link="joint3"/>
    <child link="joint4"/>
    <axis xyz="0 0 1"/>
    <limit effort="100" lower="-2.35619449019" upper="2.35619449019" velocity="5.0"/>
    <dynamics damping="0.7"/>
  </joint>
  <link name="joint4">
    <collision>
      <origin rpy="0.0 0.0 0.0" xyz="0 0.0 0"/>
      <geometry>
        <cylinder length="0.2" radius="0.05"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 -0.0 0" xyz="0 0.0  0"/>
      <geometry>
        <cylinder length="0.2" radius="0.05"/>
      </geometry>
    </visual>
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 0.00000 0.0"/>
      <mass value="0.5"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <!--    <joint name="link4_fix" type="fixed">
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <parent link="joint4"/>
        <child link="link4"/>
    </joint>
    <link name="link4">
        <collision>
            <origin xyz="0 0 ${l4/2}" rpy="0 0 0"/>
            <geometry>
                <box size="${width} ${width} ${l4} " />
            </geometry>
        </collision>
        <visual>
            <origin xyz="0 0 ${l4/2}" rpy="0 0 0"/>
            <geometry>
                <box size="${width} ${width} ${l4} "/>
            </geometry>
            <material name="darkgreen"/>
        </visual>
        <inertial>
            <origin xyz="0.0 0.00000 0.0" rpy="0 0 0"/>
            <mass value="0.5"/>
            <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0" />
        </inertial>
    </link>
    -->
  <joint name="j5" type="revolute">
    <origin rpy="-1.57079632679 3.14159265359 0.0" xyz="0 0 0.2"/>
    <parent link="joint4"/>
    <child link="joint5"/>
    <axis xyz="0 0 1"/>
    <limit effort="100" lower="-2.35619449019" upper="2.35619449019" velocity="5.0"/>
    <dynamics damping="0.7"/>
  </joint>
  <link name="joint5">
    <collision>
      <origin rpy="0.0 0.0 0.0" xyz="0 0.0 0"/>
      <geometry>
        <cylinder length="0.2" radius="0.05"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 -0.0 0" xyz="0 0.0  0"/>
      <geometry>
        <cylinder length="0.2" radius="0.05"/>
      </geometry>
    </visual>
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 0.00000 0.0"/>
      <mass value="0.5"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <!--    <joint name="link5_fix" type="fixed">
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <parent link="joint5"/>
        <child link="link5"/>
    </joint>
    <link name="link5">
        <collision>
            <origin xyz="0 ${l5/2} 0" rpy="0 0 0"/>
            <geometry>
                <box size="${width} ${l5} ${width}" />
            </geometry>
        </collision>
        <visual>
            <origin xyz="0 ${l5/2} 0" rpy="0 0 0"/>
            <geometry>
                <box size="${width} ${l5}  ${width}"/>
            </geometry>
            <material name="green"/>
        </visual>
        <inertial>
            <origin xyz="0.0 0.00000 0.0" rpy="0 0 0"/>
            <mass value="0.5"/>
            <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0" />
        </inertial>
    </link>
    -->
  <joint name="j6" type="revolute">
    <origin rpy="-1.57079632679 3.14159265359 0.0" xyz="0 0.2 0"/>
    <parent link="joint5"/>
    <child link="joint6"/>
    <axis xyz="0 0 1"/>
    <limit effort="100" lower="-2.35619449019" upper="2.35619449019" velocity="5.0"/>
    <dynamics damping="0.7"/>
  </joint>
  <link name="joint6">
    <collision>
      <origin rpy="0.0 0.0 0.0" xyz="0 0.0 0"/>
      <geometry>
        <cylinder length="0.2" radius="0.05"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 -0.0 0" xyz="0 0.0  0"/>
      <geometry>
        <cylinder length="0.2" radius="0.05"/>
      </geometry>
    </visual>
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 0.00000 0.0"/>
      <mass value="0.5"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <!--
    <joint name="link6_fix" type="fixed">
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <parent link="joint6"/>
        <child link="link6"/>
    </joint>
    <link name="link6">
        <collision>
            <origin xyz="0 0 ${l6}" rpy="0 0 0"/>
            <geometry>
                <sphere radius="${end_ball_radius}" />
            </geometry>
        </collision>
        <visual>
            <origin xyz="0 0 ${l6}" rpy="0 0 0"/>
            <geometry>
                <sphere radius="${end_ball_radius}" />
            </geometry>
            <material name="blue"/>
        </visual>
        <inertial>
            <origin xyz="0.0 0.00000 0.0" rpy="0 0 0"/>
            <mass value="0.5"/>
            <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0" />
        </inertial>
    </link>
    
    -->
  <gazebo reference="link1">
    <material>Gazebo/DarkGrey</material>
  </gazebo>
  <gazebo reference="link2">
    <material>Gazebo/FlatBlack</material>
  </gazebo>
  <gazebo reference="link3">
    <material>Gazebo/DarkGrey</material>
  </gazebo>
  <gazebo reference="link4">
    <material>Gazebo/FlatBlack</material>
  </gazebo>
  <gazebo reference="link5">
    <material>Gazebo/DarkGrey</material>
  </gazebo>
  <gazebo reference="link6">
    <material>Gazebo/FlatBlack</material>
  </gazebo>
  <gazebo reference="link7">
    <material>Gazebo/DarkGrey</material>
  </gazebo>
  <gazebo reference="end_ball">
    <material>Gazebo/RedGlow</material>
  </gazebo>
  <!--
    <transmission name="transmission">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="j1">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor1">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="transmission">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="j2">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor2">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="transmission">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="j3">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor3">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="transmission">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="j4">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor4">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="transmission">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="j5">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor5">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="transmission">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="j6">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor6">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    
    
    
    <gazebo>
        <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
            <robotNamespace>/my_robotic_arm</robotNamespace>
        </plugin>
    </gazebo>-->
</robot>

